contrib/jackc/pgx/v5/trace.go

```diff
@@ -0,0 +1,184 @@
+		ddtracer.Tag(ext.Component, componentName),
+		ddtracer.Tag(ext.SpanKind, ext.SpanKindClient),
+	}
+}
+
+func (t *tracer) spanOptions(opts ...ddtrace.StartSpanOption) []ddtrace.StartSpanOption {
+	return append(t.defaultSpanOptions(), opts...)
+}
+
+func finish(ctx context.Context, err error) {
+	span, exists := ddtracer.SpanFromContext(ctx)
+	if !exists {
+		return
+	}
+
+	if err != nil {
+		span.SetTag(ext.Error, err)
+	}
+
+	span.Finish()
+}
+
+func (t *tracer) TraceQueryStart(ctx context.Context, _ *pgx.Conn, data pgx.TraceQueryStartData) context.Context {
+	opts := t.spanOptions(
+		ddtracer.StartTime(time.Now()),
+		ddtracer.ResourceName(data.SQL),
+	)
+
+	_, ctx = ddtracer.StartSpanFromContext(ctx, "pgx.query", opts...)
+
+	return ctx
+}
+
+func (t *tracer) TraceQueryEnd(ctx context.Context, _ *pgx.Conn, data pgx.TraceQueryEndData) {
+	finish(ctx, data.Err)
+}
+
+func (t *tracer) TraceBatchStart(ctx context.Context, _ *pgx.Conn, _ pgx.TraceBatchStartData) context.Context {
+	if !t.traceBatch {
+		return ctx
+	}
+
+	opts := t.spanOptions(ddtracer.StartTime(time.Now()))
+
+	_, ctx = ddtracer.StartSpanFromContext(ctx, "pgx.batch", opts...)
+
+	return ctx
+}
+
+func (t *tracer) TraceBatchQuery(ctx context.Context, _ *pgx.Conn, data pgx.TraceBatchQueryData) {
+	if !t.traceBatch {
+		return
+	}
+
+	opts := t.spanOptions(
+		ddtracer.StartTime(time.Now()),
+		ddtracer.ResourceName(data.SQL),
+	)
+
+	_, ctx = ddtracer.StartSpanFromContext(ctx, "pgx.batch.query", opts...)
+
+	finish(ctx, data.Err)
+}
+
+func (t *tracer) TraceBatchEnd(ctx context.Context, _ *pgx.Conn, data pgx.TraceBatchEndData) {
+	if !t.traceBatch {
+		return
+	}
+
+	finish(ctx, data.Err)
+}
+
+func (t *tracer) TraceCopyFromStart(ctx context.Context, _ *pgx.Conn, data pgx.TraceCopyFromStartData) context.Context {
+	if !t.traceCopyFrom {
+		return ctx
+	}
+
+	opts := t.spanOptions(
+		ddtracer.StartTime(time.Now()),
+		ddtracer.Tag("tables", data.TableName),
+		ddtracer.Tag("columns", data.ColumnNames),
+	)
+
+	_, ctx = ddtracer.StartSpanFromContext(ctx, "pgx.copyfrom", opts...)
+
+	return ctx
+}
+
+func (t *tracer) TraceCopyFromEnd(ctx context.Context, _ *pgx.Conn, data pgx.TraceCopyFromEndData) {
+	if !t.traceCopyFrom {
+		return
+	}
+
+	finish(ctx, data.Err)
+}
+
+func (t *tracer) TracePrepareStart(ctx context.Context, _ *pgx.Conn, data pgx.TracePrepareStartData) context.Context {
+	if !t.tracePrepare {
+		return ctx
+	}
+
+	opts := t.spanOptions(
+		ddtracer.StartTime(time.Now()),
+		ddtracer.ResourceName(data.SQL),
+	)
+
+	_, ctx = ddtracer.StartSpanFromContext(ctx, "pgx.prepare", opts...)
+
+	return ctx
+}
+
+func (t *tracer) TracePrepareEnd(ctx context.Context, _ *pgx.Conn, data pgx.TracePrepareEndData) {
+	if !t.tracePrepare {
+		return
+	}
+
+	finish(ctx, data.Err)
+}
+
+func (t *tracer) TraceConnectStart(ctx context.Context, _ pgx.TraceConnectStartData) context.Context {
+	if !t.traceConnect {
+		return ctx
+	}
+
+	opts := t.spanOptions(
```

Comment:
The now time is the default so no need to create and use this option here

Reply:
Removed the start time from all span options.